projects/button_adc/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void prv_adc_callback(const GpioAddress *address, void *context) {
+  AdcChannel *analogChannel = context;
+  uint16_t analogPinData = 0;
+  adc_read_raw(*analogChannel, &analogPinData);
+  LOG_DEBUG("Analog Pin data: %d\n", analogPinData);

```

Comment:
Please use `snake_case`, not `camelCase`. This goes for the rest of the camelCase occurrences too.